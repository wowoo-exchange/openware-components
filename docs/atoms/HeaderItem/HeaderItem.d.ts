import * as React from 'react';
interface HeaderItemProps {
    /**
     * Additional class name for styling. By default element receives `cr-header-item` class
     * @default empty
     */
    className?: string;
    /**
     * The string to use as the label for the HeaderItem.
     */
    label: string;
    /**
     * Number to use as a summary
     */
    amount: number;
    /**
     * Optinal string that define currency if it needs
     * @default empty
     */
    currency?: string;
    /**
     * Optinal sign that define change if it needs
     * @default empty
     */
    sign?: string;
}
declare const HeaderItem: React.FunctionComponent<HeaderItemProps>;
export { HeaderItem, HeaderItemProps, };
