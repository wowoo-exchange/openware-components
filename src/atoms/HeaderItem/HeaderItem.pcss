.cr-header-item {
  align-content: left;

  &__label {
    font-size: 16px;
    font-style: regular;
    line-height: 19px;
    color: #657693;
  }

  &__amount-plus {
    font-weight: 500;
    font-size: 18px;
    font-style: bold;
    line-height: 21px;
    color: #65c789;
  }

  &__amount-minus {
    font-weight: 500;
    font-size: 18px;
    font-style: bold;
    line-height: 21px;
    color: #e95b5b;
  }

  &__amount-default {
    font-weight: 500;
    font-size: 18px;
    font-style: bold;
    line-height: 21px;
    color: #c1d4df;
  }
}
