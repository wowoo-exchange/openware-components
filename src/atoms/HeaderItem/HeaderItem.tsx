import classnames from 'classnames';
import * as React from 'react';

interface HeaderItemProps {
    /**
     * Additional class name for styling. By default element receives `cr-header-item` class
     * @default empty
     */
    className?: string;
    /**
     * The string to use as the label for the HeaderItem.
     */
    label: string;
    /**
     * Number to use as a summary
     */
    amount: number;
    /**
     * Optinal string that define currency if it needs
     * @default empty
     */
    currency?: string;
    /**
     * Optinal sign that define change if it needs
     * @default empty
     */
    sign?: string;
}

const HeaderItem: React.FunctionComponent<HeaderItemProps> = props => {
    const {
      label,
      amount,
      className,
      currency,
      sign,
    } = props;
    const cx = classnames('cr-header-item', className);

    const cxAmount = classnames({
        'cr-header-item__amount-plus': sign === '+' && currency === undefined,
        'cr-header-item__amount-minus': sign === '-' && currency === undefined,
        'cr-header-item__amount-default': sign === undefined && currency !== undefined,
    });

    return (
        <div className={cx}>
          {currency ? <div className={cxAmount}>{amount} {currency.toUpperCase()}</div> : <div className={cxAmount}>{sign}{amount}{sign ? '%' : ''}</div>}
          <div className="cr-header-item__label">{label}</div>
        </div>
    );
};

export {
    HeaderItem,
    HeaderItemProps,
};
