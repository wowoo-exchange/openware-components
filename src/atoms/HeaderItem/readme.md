Example of usage:

```js
  <HeaderItem label="24 hour price" amount={10.05} currency="btc" />
```