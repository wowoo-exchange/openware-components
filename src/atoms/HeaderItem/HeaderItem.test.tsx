import { shallow } from 'enzyme';
import * as React from 'react';
import { HeaderItem, HeaderItemProps } from './HeaderItem';

const defaultProps: HeaderItemProps = {
    amount: 12.1,
    label: '24 hour price',
    currency: 'btc',
};

const setup = (props: Partial<HeaderItemProps> = {}) =>
    shallow(<HeaderItem {...{ ...defaultProps, ...props }} />);

describe('HeaderItem', () => {
    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        const wrapper = setup();
        expect(wrapper.hasClass('cr-header-item')).toBeTruthy();
    });

    it('should pass along supplied className', () => {
        const className = 'some-class';
        const wrapper = setup({ className });
        expect(wrapper.hasClass(className)).toBeTruthy();
    });

    it('should render correct label', () => {
        const wrapper = setup();
        const { className } = wrapper.find('div').last().props();
        expect(className).toContain('cr-header-item__label');
    });

    it('should render correctly with currency', () => {
        const wrapper = setup();
        expect(wrapper.find('div').get(1).props.className)
            .toContain('cr-header-item__amount-default');
    });

    it('should render correctly with plus sign', () => {
        const wrapper = setup({currency: undefined, sign: '+'});
        expect(wrapper.find('div').get(1).props.className)
            .toContain('cr-header-item__amount-plus');
    });

    it('should render correctly with minus sign', () => {
        const wrapper = setup({currency: undefined, sign: '-'});
        expect(wrapper.find('div').get(1).props.className)
            .toContain('cr-header-item__amount-minus');
    });

    it('should render correctly without currency', () => {
        const wrapper = setup({currency: undefined});
        expect(wrapper.props().children[0].props.children[0]).toEqual(undefined);
    });

});
