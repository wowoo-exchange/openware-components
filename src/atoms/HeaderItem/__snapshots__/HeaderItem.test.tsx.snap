// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderItem should render 1`] = `
<div
  className="cr-header-item"
>
  <div
    className="cr-header-item__amount-default"
  >
    12.1
     
    BTC
  </div>
  <div
    className="cr-header-item__label"
  >
    24 hour price
  </div>
</div>
`;
